discord-bot/bot.py

```diff
@@ -8,5 +22,79 @@ async def on_ready():
     print('We have logged in as {0.user}'.format(client))
 
 
+@client.event
+async def on_message(message):
+    bot_id = client.user.id
+    if str(bot_id) in message.content:
+        option = str(message.content).split(f"<@!{bot_id}> ")[1].split(",")[0]
+
+        if option == "view":
+            pass
+
+        elif option == "rank":
+            data = requests.get(GET_TOP_FELLOWS)
+            top_fellows = ""
+            for i in range(len(data.json()["data"])):
+                top_fellows += f"{i+1} - {data.json()['data'][i]['name']}\n"
+            await message.channel.send(top_fellows)
+
+        elif option == "add":
+            # only admin
+            if get(message.author.roles, name="admin"):
+                splitted_msg = str(message.content).split(f"<@!{bot_id}> ")[1].split(",")
```

Comment:
Do we want to have a error handling if the user doesn't provide a correctly formed request? For example `add "I did a great job!" 10`

Reply:
Done